bika/lims/upgrade/v01_02_005.py

```diff
@@ -43,6 +47,18 @@ def upgrade(tool):
 
     return True
 
+
+def fix_permission_on_analysisrequests():
+    catalog = api.get_tool(CATALOG_ANALYSIS_REQUEST_LISTING)
+    brains = catalog(portal_type='AnalysisRequest')
```

Comment:
Apply these changes only to ARs for which `verify` transition has not been performed yet. Also exclude those that are not active. These modifs will make the upgrade step faster in instances with large amount of ARs. Something like:
```python
valid_states = ['sample_due', 'sample_received', 'sampled', 'to_be_sampled', 'to_be_preserved']
brains = catalog(cancellation_state='active', review_state=valid_states)
```
Note that there is no need to filter by portal_type (the analysis requests catalog only contains ARs).